app/src/main/kotlin/com/mapbox/vision/teaser/MainActivity.kt

```diff
@@ -1,55 +1,531 @@
+import com.mapbox.vision.mobile.core.models.road.LaneEdgeType
+import com.mapbox.vision.mobile.core.models.road.RoadDescription
+import com.mapbox.vision.mobile.core.utils.SystemInfoUtils
+import com.mapbox.vision.performance.ModelPerformance
+import com.mapbox.vision.performance.ModelPerformanceMode
+import com.mapbox.vision.performance.ModelPerformanceRate
+import com.mapbox.vision.teaser.view.show
 import com.mapbox.vision.safety.VisionSafetyManager
 import com.mapbox.vision.teaser.MainActivity.VisionManagerMode.Camera
 import com.mapbox.vision.teaser.MainActivity.VisionManagerMode.Replay
+import com.mapbox.vision.safety.core.VisionSafetyListener
+import com.mapbox.vision.safety.core.models.CollisionDangerLevel
+import com.mapbox.vision.safety.core.models.CollisionObject
+import com.mapbox.vision.safety.core.models.RoadRestrictions
 import com.mapbox.vision.teaser.ar.ArMapActivity
+import com.mapbox.vision.teaser.models.UiSign
 import com.mapbox.vision.teaser.recorder.RecorderFragment
 import com.mapbox.vision.teaser.replayer.ArReplayNavigationActivity
 import com.mapbox.vision.teaser.replayer.ReplayModeFragment
-import com.mapbox.vision.teaser.view.BaseTeaserActivity
+import com.mapbox.vision.teaser.utils.SoundsPlayer
+import com.mapbox.vision.teaser.utils.classification.SignResources
+import com.mapbox.vision.teaser.utils.classification.Tracker
+import com.mapbox.vision.teaser.utils.dpToPx
 import com.mapbox.vision.teaser.view.hide
-import com.mapbox.vision.teaser.view.show
+import com.mapbox.vision.utils.VisionLogger
 import com.mapbox.vision.view.VisionView
 import com.mapbox.vision.view.VisualizationMode
 import kotlinx.android.synthetic.main.activity_main.*
 
-class MainActivity : BaseTeaserActivity(), ReplayModeFragment.OnSelectModeItemListener {
+class MainActivity : AppCompatActivity(), ReplayModeFragment.OnSelectModeItemListener {
 
     enum class VisionManagerMode {
         Camera,
         Replay,
     }
 
-    private var mode = Camera
+    enum class VisionMode {
+        Segmentation,
+        Classification,
+        Detection,
+        Safety,
+        Lanes,
+    }
+
+    companion object {
+        private val BASE_SESSION_PATH = "${Environment.getExternalStorageDirectory().absolutePath}/MapboxVisionTelemetry"
+        private const val PERMISSION_FOREGROUND_SERVICE = "android.permission.FOREGROUND_SERVICE"
+        private const val PERMISSIONS_REQUEST_CODE = 123
+    }
+
+    private var visionManagerMode = Camera
     private var sessionPath = ""
 
-    override fun initViews(root: View) {
-        root.findViewById<LinearLayout>(R.id.ar_navigation_button_container).apply {
-            setOnClickListener {
-                when (mode) {
-                    Camera -> startActivity(Intent(this@MainActivity, ArMapActivity::class.java))
-                    Replay -> startArSession()
+    private val signResources: SignResources = SignResources.Impl(this)
+
+    private var signSize = 0
```

Comment:
```suggestion
    private val signSize = dpToPx(64f).toInt()
```
not sure, that it's necessary only when we get permissions

Reply:
fixed

Reply:
rolled back, context doen't have valid pointer to resources during init